e2e_tests/tests/requirements.txt

```diff
@@ -1,14 +1,15 @@
 appdirs
 # TODO(MLG-336): remove this requirement when tensorflow-macos is upgraded.
```

Comment:
I assume this TODO can be resolved.

The new version restrictions might also be worth a short comment with reasoning.

Reply:
Could you please add a short comment with the reasoning behind the pin for us to know later on if and when it should be touched? It's at the moment not clear to me if it wants the "darwin, etc." conditional.

Reply:
The conditional is important for distinguishing MacOS machines. I don't think we want to mess with that. Switching from `tensorflow-macos` to `tensorflow-metal` will be done later and separately.

Reply:
New comment/pin look good.